dbt_integration.py

```diff
@@ -558,6 +571,38 @@ def get_columns(self, node: "ManifestNode") -> List["ColumnInfo"]:
             columns.extend(result.table.column_names)
         return columns
 
+    def get_catalog(self) -> Dict[str, Any]:
+        """Get catalog from adapter"""
+        catalog_table: agate.Table = agate.Table([])
+        catalog_data: List[PrimitiveDict] = []
+        exceptions: List[Exception] = []
+        logger.debug("getting connection")
+        try:
+            with self.adapter.connection_named("generate_catalog"):
+                logger.debug("Generating catalog...")
+                catalog_table, exceptions = self.adapter.get_catalog(self.dbt)
+                logger.debug("Raw catalog:")
+                logger.debug(catalog_table)
+
+            if exceptions:
+                logger.debug("Found exceptions")
+                logger.debug(exceptions)
+
+            catalog_data = [
+                dict(
+                    zip(catalog_table.column_names, map(dbt.utils._coerce_decimal, row))
+                )
+                for row in catalog_table
+            ]
+            logger.debug("parsed catalog data")
+            logger.debug(catalog_data)
+
+        except Exception as e:
+            logger.debug("exception during get catalog event")
+            logger.debug(e)
+            return catalog_data
```

Comment:
Shouldn't we expose the exception as a string to the user?

Reply:
TODO for a future page where we can select a project before scanning.